services/headless-lms/models/src/exercise_task_submissions.rs

```diff
@@ -0,0 +1,368 @@
+        submission_data.data_json,
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(res.id)
+}
+
+pub async fn insert(
+    conn: &mut PgConnection,
+    exercise_slide_submission_id: Uuid,
+    exercise_slide_id: Uuid,
+    exercise_task_id: Uuid,
+    data_json: Value,
+) -> ModelResult<Uuid> {
+    let res = sqlx::query!(
+        "
+INSERT INTO exercise_task_submissions (
+    exercise_slide_submission_id,
+    exercise_slide_id,
+    exercise_task_id,
+    data_json
+  )
+  VALUES ($1, $2, $3, $4)
+  RETURNING id
+",
+        exercise_slide_submission_id,
+        exercise_slide_id,
+        exercise_task_id,
+        data_json,
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(res.id)
+}
+
+pub async fn get_by_id(conn: &mut PgConnection, id: Uuid) -> ModelResult<ExerciseTaskSubmission> {
+    let submission = sqlx::query_as!(
+        ExerciseTaskSubmission,
+        "
+SELECT *
+FROM exercise_task_submissions
+WHERE id = $1
+",
+        id
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok(submission)
+}
+
+pub async fn get_by_exercise_slide_submission_id(
+    conn: &mut PgConnection,
+    exercise_slide_submission_id: Uuid,
+) -> ModelResult<Vec<ExerciseTaskSubmission>> {
+    let submissions = sqlx::query_as!(
+        ExerciseTaskSubmission,
+        "
+SELECT *
+FROM exercise_task_submissions
+WHERE exercise_slide_submission_id = $1
+  AND deleted_at IS NULL
+        ",
+        exercise_slide_submission_id
+    )
+    .fetch_all(conn)
+    .await?;
+    Ok(submissions)
+}
+
+pub async fn get_users_latest_exercise_task_submissions_for_exercise_slide(
+    conn: &mut PgConnection,
+    exercise_slide_id: &Uuid,
+    user_id: &Uuid,
+) -> ModelResult<Option<Vec<ExerciseTaskSubmission>>> {
+    let exercise_slide_submission =
+        exercise_slide_submissions::get_users_latest_exercise_slide_submission(
+            conn,
+            exercise_slide_id,
+            user_id,
+        )
+        .await?;
+    if let Some(exercise_slide_submission) = exercise_slide_submission {
+        let task_submissions = sqlx::query_as!(
+            ExerciseTaskSubmission,
+            "
+SELECT *
+FROM exercise_task_submissions
+WHERE exercise_slide_submission_id = $1
+  AND deleted_at IS NULL
+            ",
+            exercise_slide_submission.id
+        )
+        .fetch_all(conn)
+        .await?;
+        Ok(Some(task_submissions))
+    } else {
+        Ok(None)
+    }
+}
+
+pub async fn get_course_and_exam_id(
+    conn: &mut PgConnection,
+    id: Uuid,
+) -> ModelResult<(Option<Uuid>, Option<Uuid>)> {
+    let res = sqlx::query!(
+        "
+SELECT ess.course_id,
+  ess.exam_id
+FROM exercise_task_submissions ets
+  JOIN exercise_slide_submissions ess ON ets.exercise_slide_submission_id = ess.id
+WHERE ets.id = $1
+  AND ets.deleted_at IS NULL
+  AND ess.deleted_at IS NULL
+        ",
+        id
+    )
+    .fetch_one(conn)
+    .await?;
+    Ok((res.course_id, res.exam_id))
+}
+
+pub async fn insert_submission(
```

Comment:
Comment that assumes we have already inserted the slide submission?

`create_task_submission_for_slide`